File: Longest_Palindromic_Substring.js
/**Given a string s, return the longest palindromic substring in s. **/

function longestPalindrome(s) {
  let longest = "";
  for (let i = 0; i < s.length; i++) {
    for (let j = i + 1; j <= s.length; j++) {
      let substr = s.substring(i, j);
      if (substr === substr.split("").reverse().join("") && substr.length > longest.length) {
        longest = substr;
      }
    }
  }
  return longest;
}
console.log(longestPalindrome("babad")); // "bab" or "aba"
console.log(longestPalindrome("cbbd")); // "bb"

File: Longest_Palindromic_Substring.py
# Given a string s, return the longest palindromic substring in s.
def longest_palindrome(s):
    longest = ""
    for i in range(len(s)):
        for j in range(i + 1, len(s) + 1):
            substr = s[i:j]
            if substr == substr[::-1] and len(substr) > len(longest):
                longest = substr
    return longest

print(longest_palindrome("babad")) # "bab" or "aba"
print(longest_palindrome("cbbd")) # "bb"
